app/models/user.rb

```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  validates :uid, presence: true
```

Comment:
I think this will fail that User can create an account manually since  a uid will not be present in creating a manual account. Should I make it so that uid = nil or some default value if it's a manual creation?

Reply:
good catch, when I wrote this I wasn't thinking about manual login.  I'll fix it now and push up the change. I'll remove the validation.  Are you thinking to default the value when it's created in the db?

Reply:
Yeah if really have to have uid present then I would do like a before_create if uid is not present then create a default value and set to uid. But If we don't need uid validation then I wouldn't need to do that.